glusterd2/commands/volumes/brick-replace.go

```diff
@@ -0,0 +1,295 @@
+package volumecommands
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"sort"
+	"strings"
+
+	"github.com/gluster/glusterd2/glusterd2/brick"
+	"github.com/gluster/glusterd2/glusterd2/bricksplanner"
+	"github.com/gluster/glusterd2/glusterd2/gdctx"
+	"github.com/gluster/glusterd2/glusterd2/peer"
+	restutils "github.com/gluster/glusterd2/glusterd2/servers/rest/utils"
+	"github.com/gluster/glusterd2/glusterd2/store"
+	"github.com/gluster/glusterd2/glusterd2/transaction"
+	"github.com/gluster/glusterd2/glusterd2/volume"
+	"github.com/gluster/glusterd2/pkg/api"
+	gderrors "github.com/gluster/glusterd2/pkg/errors"
+	deviceapi "github.com/gluster/glusterd2/plugins/device/api"
+	"github.com/gluster/glusterd2/plugins/device/deviceutils"
+
+	"github.com/gorilla/mux"
+	"github.com/pborman/uuid"
+)
+
+func registerReplaceBrickStepFuncs() {
+	var sfs = []struct {
+		name string
+		sf   transaction.StepFunc
+	}{
+		{"brick-repalce.txnPrepareBrick", txnPrepareBrick},
+		{"brick-replace.ReplaceVolinfo", replaceVolinfo},
+		{"brick-replace.StartBrick", startBrick},
+	}
+	for _, sf := range sfs {
+		transaction.RegisterStepFunc(sf.sf, sf.name)
+	}
+}
+
+func replaceBrickHandler(w http.ResponseWriter, r *http.Request) {
+
+	ctx := r.Context()
+
+	volname := mux.Vars(r)["volname"]
+
+	// Get Volume Info
+	vol, err := volume.GetVolume(volname)
+	if err != nil {
+		status, err := restutils.ErrToStatusCode(err)
+		restutils.SendHTTPError(ctx, w, status, err)
+		return
+	}
+
+	// Unmarshal ReplaceBrickReq
+	var req api.ReplaceBrickReq
+	if err := restutils.UnmarshalRequest(r, &req); err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, gderrors.ErrJSONParsingFailed)
+		return
+	}
+
+	subVols := vol.Subvols
+	var peerIDs []uuid.UUID
+	var srcBrickInfo brick.Brickinfo
+	m := make(map[string]bool)
+	brickCount := 0
+	subVolIndex := 0
+	brickIndex := 0
+
+	for index := range subVols {
+		for i, brick := range subVols[index].Bricks {
+			brickCount += 1
+			// Get Source Brick Info
+			if brick.PeerID.String() == req.SrcPeerID && brick.Path == req.SrcBrickPath {
+				subVolIndex = index
+				brickIndex = i
+				srcBrickInfo = brick
+			}
+			// Avoid duplicate appends
+			if _, ok := m[brick.PeerID.String()]; !ok {
+				m[brick.PeerID.String()] = true
+				peerIDs = append(peerIDs, brick.PeerID)
+			}
+		}
+	}
+
+	/*if srcBrickInfo == (brick.Brickinfo{}) {
+		//Brick not found, quit with error msg
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, "source brick not found")
+		return
+	}*/
+
```

Comment:
Please remove comments.

Reply:
I would like to keep these for the time being, I am working on a small feature which requires these lines of code.